lib/pbench/server/api/resources/upload_api.py

```diff
@@ -60,6 +61,16 @@ def __init__(self, config, logger):
         self.temporary.mkdir(mode=0o755, parents=True, exist_ok=True)
         self.logger.info("Configured PUT temporary directory as {}", self.temporary)
 
+    def drain(self, stream: IO[bytes]):
+        self.logger.info("Draining input stream")
+        total = 0
+        while True:
+            chunk = stream.read(self.CHUNK_SIZE)
+            if len(chunk) == 0:
+                break
+            total += len(chunk)
```

Comment:
Alternately:
```
        length = 1
        while length > 0:
            length = len(stream.read(self.CHUNK_SIZE))
            total += length
```
This is one line shorter and it calculates the length only once per iteration.  (And, it avoids the `while True` construction, which I dislike.  😉)

Or, possibly,
```
        while (length := len(stream.read(self.CHUNK_SIZE))) > 0:
            total += length
```
This avoids the "priming the pump" assignment.

Reply:
Maybe, but I don't mind that `True` exists, and I don't really see a point to changing it unless something else forces me to mung this file again.